service/routes.py

```diff
@@ -510,6 +534,64 @@ def get(self, shopcart_id, item_id):
         item_js = item.serialize()
         return item_js, status.HTTP_200_OK
 
+        @api.doc("put_items")
+    @api.response(404, "Shopcart or Item not found")
+    @api.response(204, "Item deleted")
+    @api.response(400, "The posted Item data was not valid")
+    @api.response(415, "Invalid header content-type")
+    @api.expect(item_model)
+    @api.marshal_with(item_model)
+    def put(self, shopcart_id, item_id):
+        """
+        Update a Item
+
+        This endpoint returns just a item
+        """
+        check_content_type(DEFAULT_CONTENT_TYPE)
+        app.logger.info("Request update item with shopcart_id: %s and item_id: %s", shopcart_id, item_id)
+        shopcart = Shopcart.get_by_id(shopcart_id)
+        if not shopcart:
+            abort(
+                status.HTTP_404_NOT_FOUND,
+                f"Shopcart with id '{shopcart_id}' could not be found."
+            )
+        item = Item.get_by_id(item_id)
+        if not item:
+            abort(
+                status.HTTP_404_NOT_FOUND,
+                f"Item with id '{item_id}' could not be found."
+            )
+        data = api.payload
+        if len(data) != 5:
+            app.logger.info("Must update the information about this item")
+            abort(
+                status.HTTP_400_BAD_REQUEST,
+                "Missing information."
+            )
+        if data["quantity"] < 0:
+            app.logger.info("Can not update item with negative quantity")
+            abort(
+                status.HTTP_400_BAD_REQUEST,
+                "Quantity of the item must be positive."
+            )
+        if data["price"] < 0:
+            app.logger.info("Can not update item with negative price")
+            abort(
+                status.HTTP_400_BAD_REQUEST,
+                "Price of the item must be positive."
+            )
+        if data["quantity"] == 0:
```

Comment:
Thank you for catching the corner case where `quantity == 0` here! However, I feel that it would be strange if we allow setting quantity to 0 🤔 because that means after an update item call an item could disappear from the database.

Reply:
Hence, I'm thinking if we can just response with **400 Bad Request** when `quantity <= 0`?

Reply:
btw, can we just copy the original update_items API logic into class and add one more check for `quantity <= 0`? otherwise, the existing test cases do not cover some of these newly added logic lol